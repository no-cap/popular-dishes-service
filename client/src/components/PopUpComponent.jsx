import React from 'react';
import PhotoBox from './PhotoBox.jsx';
import ReviewsBox from './ReviewsBox.jsx';
import { 
  SmallDescriptionFormatter,
  SmallDescriptionBorder, 
  ItemDescription, 
  ReviewsFormatter,
  Name,
  Price
} from '../components-style/PopupComponent-style.jsx';

class PopUpComponent extends React.Component {
  constructor(props) {
    super(props)
    this.photoBoxElement = React.createRef();
    this.resetPhotoBox = this.resetPhotoBox.bind(this);
    this.handleKeyPress = this.props.handleKeyPress;
  }

  resetPhotoBox() {
    this.photoBoxElement.current.resetCurrent();
  }

  componentDidMount() {
    document.addEventListener("keydown", this.props.handleKeyPress);
  }

  componentWillUnmount() {
    document.removeEventListener("keydown", this.props.handleKeyPress, false);
  }

  render() {
    return (
      <ItemDescription>
        <PhotoBox photos={this.props.photos} ref={this.photoBoxElement} />
        <SmallDescriptionFormatter>
            <Name>{this.props.item.dish_name}</Name>
            <Price>${this.props.item.price}.00</Price>
        </SmallDescriptionFormatter>
        <ReviewsFormatter>
          <SmallDescriptionBorder> <p>{this.props.item.description}</p> </SmallDescriptionBorder>
          <ReviewsBox reviews={this.props.reviews} />
        </ReviewsFormatter>
      </ItemDescription>

    )
  }
}

export default PopUpComponent;
