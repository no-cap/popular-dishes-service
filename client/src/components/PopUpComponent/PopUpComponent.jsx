import React from 'react';
import PhotoBox from '../PhotoBox/PhotoBox';
import ReviewsBox from '../ReviewsBox/ReviewsBox';
import {
  SmallDescriptionFormatter,
  SmallDescriptionBorder,
  ItemDescription,
  ReviewsFormatter,
  Name,
  Price
} from './PopupComponentStyles.jsx';

class PopUpComponent extends React.Component {
  constructor(props) {
    super(props)
    this.photoBoxElement = React.createRef();
    this.resetPhotoBox = this.resetPhotoBox.bind(this);
    this.handleKeyPress = this.props.handleKeyPress;
  }

  componentDidMount() {
    document.addEventListener('keydown', this.props.handleKeyPress);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.props.handleKeyPress, false);
  }

  resetPhotoBox() {
    this.photoBoxElement.current.resetCurrent();
  }

  render() {
    const { photos, dishName, price, description, reviews } = this.props.dish;
    const { host } = this.props;
    return (
      <ItemDescription>
        <PhotoBox photos={photos} ref={this.photoBoxElement} />
        <SmallDescriptionFormatter>
          <Name>{dishName}</Name>
          <Price>{`${price}`}</Price>
        </SmallDescriptionFormatter>
        <ReviewsFormatter>
          <SmallDescriptionBorder>
            <p>{description}</p>
          </SmallDescriptionBorder>
          <ReviewsBox reviews={reviews} host={host} />
        </ReviewsFormatter>
      </ItemDescription>
    );
  }
}

export default PopUpComponent;
