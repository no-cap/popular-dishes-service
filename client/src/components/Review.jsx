import React from 'react';
import StarBox from './starbox.jsx';
import Star from './star.jsx';
import PersonIcon from './personIcon.jsx';
import StarType from '../components-style/TypesOfStars.jsx';

import { 
    SingleReview, 
    PhotoCropper, 
    ProfilePic, 
    UserName, 
    TinyPairIcon, 
    FriendCount, 
    TinyStar, 
    UserReviews,
    StarContainer,
    Date,
    ReviewText
} from '../components-style/SingleReview.jsx'

class Review extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: this.props.data.text,
            readMore: false
        };
        this.onClickHandler = this.onClickHandler.bind(this);
        this.dateFormatter = this.dateFormatter.bind(this);
    }
    // handles clicking on Read more / Read less
    onClickHandler(e) {
        this.setState({ readMore: !this.state.readMore })
    }
    // Formats Dates
    dateFormatter(date) {
        var arr = date.slice(0, 11).split("-");
        if (arr[1][0] === '0') {
            var dateWithZero = arr[1] + '/' + arr[0] + '/' + arr[2];
            return dateWithZero.slice(1);
        } else {
            return arr[1] + '/' + arr[0] + '/' + arr[2];
        }
    }

    render() {
        return (
            <SingleReview >
                <PhotoCropper> <ProfilePic src={this.props.data.userphoto} /> </PhotoCropper>
                <UserName> {this.props.data.username} </UserName>
                <TinyPairIcon > 
                    <PersonIcon /> 
                </TinyPairIcon>
                <FriendCount > {this.props.data.friends} </FriendCount>
                <TinyStar> 
                    <StarType type="mini" rating={4}> <Star/> 
                </StarType> </TinyStar>
                <UserReviews> {this.props.data.reviews} </UserReviews>
                <StarContainer> 
                    <StarBox rating={this.props.data.rating} /> 
                </StarContainer>
                <Date> {this.dateFormatter(this.props.data.date)} </Date>
                <ReviewText>
                    <div id={this.props.data.user}>{(this.state.readMore) ? this.state.text : this.state.text.split(" ").slice(0, 22).join(" ")}</div>
                    {(this.props.data.text.split(" ").length > 22) ? <button onClick={this.onClickHandler}> {(this.state.readMore) ? 'Read less' : 'Read more'} </button> : null}
                </ReviewText>
            </SingleReview>
        )
    }
}

export default Review;


// { JSON.stringify(this.props.data) }