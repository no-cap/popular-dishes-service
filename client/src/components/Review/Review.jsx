import React from 'react';
import StarBox from '../Starbox/Starbox';
import Star from '../Starbox/Star';
import PersonIcon from '../PersonIcon/PersonIcon';
import StarType from '../Starbox/StarStyles';

import {
  SingleReview,
  PhotoCropper,
  ProfilePic,
  UserName,
  TinyPairIcon,
  FriendCount,
  TinyStar,
  UserReviews,
  StarContainer,
  Date,
  ReviewText,
} from './ReviewStyles';

class Review extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: this.props.data.text,
      readMore: false,
    };
    this.onClickHandler = this.onClickHandler.bind(this);
    this.dateFormatter = this.dateFormatter.bind(this);
  }

  // handles clicking on Read more / Read less
  onClickHandler(e) {
    this.setState({ readMore: !this.state.readMore });
  }

  // Formats Dates
  // eslint-disable-next-line class-methods-use-this
  dateFormatter(date) {
    const arr = date.slice(0, 11).split('-');
    if (arr[1][0] === '0') {
      const dateWithZero = `${arr[1]}/${arr[0]}/${arr[2]}`;
      return dateWithZero.slice(1);
    }
    return `${arr[1]}/${arr[0]}/${arr[2]}`;
  }

  render() {
    return (
      <SingleReview>
        <PhotoCropper>
          <ProfilePic src={this.props.data.userphoto} />
        </PhotoCropper>
        <UserName>{this.props.data.username}</UserName>
        <TinyPairIcon>
          <PersonIcon />
        </TinyPairIcon>
        <FriendCount>{this.props.data.friends}</FriendCount>
        <TinyStar>
          <StarType type="mini" rating={4}><Star /></StarType>
        </TinyStar>
        <UserReviews>{this.props.data.reviews}</UserReviews>
        <StarContainer>
          <StarBox rating={this.props.data.rating} />
        </StarContainer>
        <Date>{this.dateFormatter(this.props.data.date)}</Date>
        <ReviewText>
          <div id={this.props.data.user}>{(this.state.readMore) ? this.state.text : this.state.text.split(' ').slice(0, 22).join(' ')}</div>
          {(this.props.data.text.split(' ').length > 22) ? (
            <button onClick={this.onClickHandler}>{(this.state.readMore) ? 'Read less' : 'Read more'}</button>
          ) : null}
        </ReviewText>
      </SingleReview>
    );
  }
}

export default Review;
