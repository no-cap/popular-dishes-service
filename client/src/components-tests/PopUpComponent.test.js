import React from 'react';
import { shallow, mount } from 'enzyme';
import PopUpComponent from '../components/PopUpComponent.jsx';
import Review from '../components/Review.jsx';
import PhotoEntry from '../components/PhotoEntry.jsx';
import SmallDescription from '../components/smallDescription.jsx';

var sampleItem = {
  name: 'Corn Cheese',
  price: 8.50,
  description: 'Corn with Cheese'
}

var sampleReviews = [
  {
    user: 'Bob',
    text: 'I love corn cheese'
  },
  {
    user: 'Bob',
    text: 'I love corn cheese'
  },
  {
    user: 'Bob',
    text: 'I love corn cheese'
  }
];

var samplePhotos = [
  {
    url: 'https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg',
    caption: 'Red Dragon roll was so good!! We devoured it in record timing!'
  },
  {
    url: 'https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg',
    caption: 'Red Dragon roll was so good!! We devoured it in record timing!'
  },
  {
    url: 'https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg',
    caption: 'Red Dragon roll was so good!! We devoured it in record timing!'
  },
  {
    url: 'https://s3-media0.fl.yelpcdn.com/bphoto/EhwQv7W3nmJqdXhvCW7pJg/o.jpg',
    caption: 'Red Dragon roll was so good!! We devoured it in record timing!'
  }
];


// all the tests
describe('PopUpComponent Unit test', () => {
  // just seeing if the component even renders correctly
  describe('PopUpComponent should render correctly', () => {
    test('PopUpComponent should render correctly in "debug" mode', () => {

      const wrapper = shallow(<PopUpComponent item={sampleItem} photos={samplePhotos} reviews={sampleReviews}/>);

      expect(wrapper).toMatchSnapshot();
    });

    test('PopUpComponent should have the same number of children as there are in their respective arrays', () => {
      const wrapper = mount(<PopUpComponent item={sampleItem} photos={samplePhotos} reviews={sampleReviews} />);
      expect(wrapper.find(Review)).toHaveLength(sampleReviews.length);
      expect(wrapper.find(PhotoEntry)).toHaveLength(1);
      expect(wrapper.find(SmallDescription)).toHaveLength(1);
    });
  });
});