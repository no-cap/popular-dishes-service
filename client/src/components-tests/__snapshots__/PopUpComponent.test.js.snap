// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpComponent Unit test PopUpComponent should render correctly PopUpComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
